### PR TITLE
[Datasource] Add available hosts

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPiAvailableHostsDataSourceBasic
--- PASS: TestAccIBMPiAvailableHostsDataSourceBasic (23.99s)
PASS

...
```

### CODE REVIEW

File: ibm/service/power/data_source_ibm_pi_available_hosts.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// Attributes
+			"available_hosts": {
```
Comment (ismirlia) [inline ibm/service/power/data_source_ibm_pi_available_hosts.go:33]: Missing constant?

Comment (michaelkad) [reply ibm/service/power/data_source_ibm_pi_available_hosts.go]: Done

File: ibm/service/power/data_source_ibm_pi_available_hosts.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+			availableHosts = append(availableHosts, map[string]interface{}{
+				Attr_Count:           int(value.Count),
+				Attr_SysType:         value.SysType,
+				Attr_AvailableCores:  value.Capacity.Cores.Total,
+				Attr_AvailableMemory: value.Capacity.Memory.Total,
+			})
+		}
+	}
+
+	d.SetId(time.Now().UTC().String())
```
Comment (ismirlia) [inline ibm/service/power/data_source_ibm_pi_available_hosts.go:86]: Is this consistent with what other data sources are doing for the id for the listing commands?

Should it be something closer to
```
	var genID, _ = uuid.GenerateUUID()
	d.SetId(genID)
```

?

File: website/docs/d/pi_available_hosts.html.markdown [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Review the argument reference that you can specify for your data source.
+
+- `pi_cloud_instance_id` - (Required, String) The GUID of the service instance associated with an account.
+
+## Attribute Reference
+
+In addition to the argument reference list, you can access the following attribute references after your data source is created.
+
+- `available_hosts` (List) Lists of all availabe host.
```
Comment (ismirlia) [inline website/docs/d/pi_available_hosts.html.markdown:49]:
```suggestion
- `available_hosts` (List) Lists of all available hosts.
```

File: ibm/service/power/data_source_ibm_pi_available_hosts.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			// Attributes
+			Attr_AvailableHosts: {
+				Computed:    true,
+				Description: "Lists of all availabe host.",
```
Comment (ismirlia) [inline ibm/service/power/data_source_ibm_pi_available_hosts.go:33]:
```suggestion
				Description: "Lists of all available hosts.",
```

File: website/docs/d/pi_available_hosts.html.markdown (hunk 3 above)
Comment (michaelkad) [reply website/docs/d/pi_available_hosts.html.markdown]: Updated.

File: ibm/service/power/data_source_ibm_pi_available_hosts.go (hunk 4 above)
Comment (michaelkad) [reply ibm/service/power/data_source_ibm_pi_available_hosts.go]: Updated.

**Review by ismirlia — APPROVED** [review]

**Comment by michaelkad** [comment]
Merged

**Pull request closed without merging**